gdx/test/com/badlogic/gdx/utils/LongArrayTest.java

```diff
@@ -0,0 +1,425 @@
+/*
+ * Copyright 2024 See AUTHORS file
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.badlogic.gdx.utils;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class LongArrayTest {
+	/** Test of the different adding methods */
+	@Test
+	public void addTest () {
+		/*
+		 * Test of the classic add(long value) method, it should be adding a number at the first available place.
+		 */
+		LongArray longArray1 = new LongArray(3);
+		longArray1.add(3);
+		Assert.assertArrayEquals(new long[] {3}, longArray1.toArray());
+
+		/*
+		 * Test of the add(long value1,long value2) method, it should be adding these numbers in the array, when it is possible.
+		 */
+		LongArray longArray2 = new LongArray();
+		longArray2.add(1, 2);
+		Assert.assertArrayEquals(new long[] {1, 2}, longArray2.toArray());
+
+		/*
+		 * Some test with the addAll(LongArray array) and addAll (long... array) methods. The second call should resize longArray3
+		 * to a size of 17.
+		 */
+		LongArray longArray3 = new LongArray();
+		longArray3.addAll(longArray2);
+		Assert.assertArrayEquals(longArray2.toArray(), longArray3.toArray());
+		longArray3.addAll(longArray1);
+		Assert.assertArrayEquals(new long[] {1, 2, 3}, longArray3.toArray());
+		longArray3.addAll(new long[] {4, 5, 6, 2, 8, 10, 1, 6, 2, 3, 30, 31, 25, 20});
+		Assert.assertEquals(17, longArray3.size);
+		Assert.assertArrayEquals(new long[] {1, 2, 3, 4, 5, 6, 2, 8, 10, 1, 6, 2, 3, 30, 31, 25, 20}, longArray3.toArray());
+
+		/*
+		 * Test of the method addAll(long[] array, int offset, int length) that adds the numbers contained between index offset to
+		 * offset+length in array to longArray4.
+		 */
+		LongArray longArray4 = new LongArray();
+		longArray4.addAll(new long[] {4, 5, 6, 2, 21, 45, 78}, 3, 3);
+		Assert.assertArrayEquals(new long[] {2, 21, 45}, longArray4.toArray());
+	}
+
+	/** Test of the get() method */
+	@Test
+	public void getTest () {
+		LongArray longArray = new LongArray();
+		longArray.add(3, 4, 5, 1);
+		Assert.assertEquals(3, longArray.get(0));
+		try {
+			longArray.get(9);
+			Assert.fail();
+		} catch (IndexOutOfBoundsException e) {
+			// We should get here as we want to get an index that's out of bounds.
+		}
+	}
+
+	/** Test of the set() method */
+	@Test
+	public void setTest () {
+		LongArray longArray = new LongArray(new long[] {3, 4, 5, 7});
```

Comment:
All these usages of the LongArray constructor that takes a `long[]` could be a good amount cleaner by using `LongArray.with(3, 4, 5, 7)` . It isn't at all needed, but I feel like with() should at least be known about, and used when possible.